string inputs
---
float64 number